db/collection.go

```diff
@@ -1082,6 +1088,108 @@ func (c *collection) save(
 	return headNode.Cid(), nil
 }
 
+func (c *collection) validateOneToOneLinkDoesntAlreadyExist(
+	ctx context.Context,
+	txn datastore.Txn,
+	docKey string,
+	fieldDescription client.FieldDescription,
+	value any,
+) error {
+	if !fieldDescription.RelationType.IsSet(client.Relation_Type_INTERNAL_ID) {
+		return nil
+	}
+
+	if value == nil {
+		return nil
+	}
+
+	objFieldDescription, ok := c.desc.Schema.GetField(strings.TrimSuffix(fieldDescription.Name, request.RelatedObjectID))
+	if !ok {
+		return client.NewErrFieldNotExist(strings.TrimSuffix(fieldDescription.Name, request.RelatedObjectID))
+	}
+	if !objFieldDescription.RelationType.IsSet(client.Relation_Type_ONEONE) {
+		return nil
+	}
+
+	fetcher := c.newFetcher()
+
+	err := fetcher.Init(ctx, txn, &c.desc, []client.FieldDescription{fieldDescription}, nil, nil, false, false)
+	if err != nil {
+		closeErr := fetcher.Close()
+		if closeErr != nil {
+			return errors.Wrap(err.Error(), closeErr)
+		}
+		return err
+	}
+
+	colPrefix := core.DataStoreKey{CollectionID: c.desc.IDString()}
+	err = fetcher.Start(ctx, core.NewSpans(core.NewSpan(colPrefix, colPrefix.PrefixEnd())))
+	if err != nil {
+		closeErr := fetcher.Close()
+		if closeErr != nil {
+			return errors.Wrap(err.Error(), closeErr)
+		}
+		return err
+	}
+
+	var alreadyLinked bool
+	var existingDocumentID string
+fetchLoop:
+	for {
+		doc, _, err := fetcher.FetchNext(ctx)
+		if err != nil {
+			closeErr := fetcher.Close()
+			if closeErr != nil {
+				return errors.Wrap(err.Error(), closeErr)
+			}
+			return err
+		}
+		if doc == nil {
+			err = fetcher.Close()
+			if err != nil {
+				return err
+			}
+			break
+		}
+
+		existingDocumentID = string(doc.Key())
+		if string(doc.Key()) == docKey {
+			continue
+		}
+
+		props, err := doc.Properties(false)
+		if err != nil {
+			closeErr := fetcher.Close()
+			if closeErr != nil {
+				return errors.Wrap(err.Error(), closeErr)
+			}
+			return err
+		}
+
+		for field, fetchedValue := range props {
```

Comment:
question: Given that you requested only one field, there should only ever be one prop, correct?

Reply:
I think at the moment all will be returned as Lens had to disable that.  Ignoring that issue, I'm not sure but _key and _deleted might be returned too but I'm not sure.

Reply:
ok thanks. Are we tracking this disabled feature?

Reply:
I don't know

Reply:
Since you had to disable it with Lens, do you mind taking care of tracking it? Having it disabled has the side effect of using a lot more memory on queries. It's important that we don't forget it.

Reply:
I was incorrect, this was changed.  Lens only disables this if there are migrations defined for the collection, and if there are there is little we can do here with the current public interfaces.  At a later date we could allow the users to control this, but there are probably more significant performance savings to be made before we get round to introducing such a thing.

Reply:
Thanks for looking into it.